exercises/concept/lasagna/src/test/java/utils/ReflectionProxy.java

```diff
@@ -0,0 +1,477 @@
+                    return c.newInstance(args);
+                } catch (Exception e) {
+                    // do nothing;
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Checks if the target class is abstract
+     * @return True if the target class exists and is abstract, false otherwise
+     */
+    public boolean isAbstract() {
+        Class<?> targetClass = getTargetClass();
+        if (targetClass == null) {
+            return false;
+        }
+        return Modifier.isAbstract(targetClass.getModifiers());
+    }
+
+    /**
+     * Checks if the target class extends another
+     * @param className The fully qualified name of the class it should extend
+     * @return True if the target class extends the specified one, false otherwise
+     */
+    public boolean extendsClass(String className) {
+        Class<?> targetClass = getTargetClass();
+        if (targetClass == null) {
+            return false;
+        }
+        try {
+            Class<?> parentClass = Class.forName(className);
+            return parentClass.isAssignableFrom(targetClass);
+        } catch (ClassNotFoundException e) {
+            return false;
+        }
+    }
+
+    /**
+     * Checks if the target class is an interface
+     * @return True if the target class exists and is an interface, false otherwise
+     */
+    public boolean isInterface() {
+        Class<?> targetClass = getTargetClass();
+        if (targetClass == null) {
+            return false;
+        }
+        return targetClass.isInterface();
+    }
+
+    /**
+     * Checks if a method is abstract
+     * @param name The name of the method
+     * @param parameterTypes The list of method parameter types
+     * @return True if the method exists and is abstract, false otherwise
+     */
+    public boolean isMethodAbstract(String name, Class<?>... parameterTypes) {
+        Class<?> targetClass = getTargetClass();
+        if (targetClass == null || name == null) {
+            return false;
+        }
+        try {
+            Method m = targetClass.getDeclaredMethod(name, parameterTypes);
+            return Modifier.isAbstract(m.getModifiers());
+        } catch (NoSuchMethodException e) {
+            return false;
+        }
+    }
+
+    /**
+     * Checks if a method is protected
+     * @param name The name of the method
+     * @param parameterTypes The list of method parameter types
+     * @return True if the method exists and is protected, false otherwise
+     */
+    public boolean isMethodProtected(String name, Class<?>... parameterTypes) {
+        Class<?> targetClass = getTargetClass();
+        if (targetClass == null || name == null) {
+            return false;
+        }
+        try {
+            Method m = targetClass.getDeclaredMethod(name, parameterTypes);
+            return Modifier.isProtected(m.getModifiers());
+        } catch (NoSuchMethodException e) {
+            return false;
+        }
+    }
+}
```

Comment:
Not all the methods are used but it seems like a good template for future use in other exercises.

Reply:
Yeah, I thought we could leave them for reference.

Reply:
I suggest to comment the unused methods with a explicit message like `Unused at moment` or something like that.
It can improve the time to understand on the code for a maintainer who unknows it

Reply:
I agree. What do you say about grouping the unused methods with a _**region**_ section? (it works for almost all editors and is still "understandable" in case it is not)

![image](https://user-images.githubusercontent.com/2152934/140362083-bc82fbf6-88e8-4234-8a78-48f537bf0ba5.png)

Reply:
It might be a good idea to create a Gradle package that contains the reflection proxy, as it can then be re-used across exercises. We've done a similar thing with C#: https://github.com/exercism/dotnet-tests

Reply:
@ErikSchierboom : that's exactly what I wanted to do, but I don't know how to setup an internal Gradle package 😅 . If you point me in the right direction I will do it 🙏

Reply:
We have to ask @iHiD to create a repository. Ideas on a good name?

Reply:
Oh, ok, I thought we could move this class to an upper folder and be able to share it with all exercises.

Naming is the most difficult part 😅 . Are there any other utility libraries to refer to, or conventions already?

If not, I would suggest something like `exercism:test-utils`

Reply:
I don't think we have any conventions. We don't yet support shared files between exercises and it might be nice to hide those files from the student anyway.

Reply:
So, should we merge this PR and create another ticket for the new repository? (are you using Jira or other issue tracking?)